packages/protocol/contracts-0.8/common/FeeCurrencyDirectory.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../../contracts/common/Initializable.sol";
+import "@openzeppelin/contracts8/access/Ownable.sol";
+
+interface IOracle {
+  function getExchangeRateFor(address identifier)
+    external
+    view
+    returns (uint256 numerator, uint256 denominator, uint256 lastUpdateTimestamp);
+}
+
+contract FeeCurrencyDirectory is Initializable, Ownable {
+  mapping(address => CurrencyConfig) public currencies;
+  address[] private currencyList;
+
+  struct CurrencyConfig {
+    address currencyIdentifier;
+    address oracle;
+    uint256 intrinsicGas;
+  }
+
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+     * @notice Initializes the contract with the owner set.
+     */
+  function initialize() public initializer {
+    _transferOwnership(msg.sender);
+  }
+
+  /**
+     * @notice Sets the currency configuration for a token.
+     * @dev This action can only be performed by the contract owner.
+     * @param token The token address.
+     * @param currencyIdentifier The currency identifier.
+     * @param oracle The oracle address for price fetching.
+     * @param intrinsicGas The intrinsic gas value for transactions.
+     */
+  function setCurrencyConfig(
```

Comment:
for later, but it's useful to have smaller setters, so governance proposals are easier to make. (would track this in an issue)

Reply:
I actually believe that it is more transparent when whole mapping is being set instead of multiple parts